apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerConnectionTests.AbortedTransactionRetryTests.cs

```diff
@@ -0,0 +1,337 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax.Testing;
+using Google.Cloud.Spanner.Common.V1;
+using Google.Cloud.Spanner.V1;
+using Google.Cloud.Spanner.V1.Internal.Logging;
+using Google.Cloud.Spanner.V1.Tests;
+using Grpc.Core;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public partial class SpannerConnectionTests
+    {
+        public sealed class TransactionAbortedRetryTests
+        {
+            // Combining backoff and jittering for retries
+            // when the RpcException doesn't specify a wait time.
+            // InitialDelay = 250ms, BackoffMultiplier = 2, JitterMultiplier = 2
+            // Those are all hardwire, client code can't change them.
+            private const int SecondCallLowBo = 250;
+            private const int SecondCallHighBo = 2 * 250;
+            private const int ThirdCallLowBo = 250 + 500;
+            private const int ThirdCallHighBo = 2 * (250 + 500);
+
+            private const int ExceptionRetryDelayMs = 300;
+            private const int SecondCallLowEx = 300;
+            private const int SecondCallHighEx = 2 * 300;
+            private const int ThirdCallLowEx = 300 + 300;
+            private const int ThirdCallHighEx = 2 * (300 + 300);
+
+            [Fact]
+            public async Task FirstCallSucceeds()
+            {
+                var spannerClientMock = SpannerClientHelpers
+                    .CreateMockClient(Logger.DefaultLogger, MockBehavior.Strict)
+                    .SetupBatchCreateSessionsAsync()
+                    .SetupBeginTransactionAsync()
+                    .SetupExecuteBatchDmlAsync()
+                    .SetupCommitAsync();
+                SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+
+                var scheduler = (FakeScheduler)connection.Builder.SessionPoolManager.SpannerSettings.Scheduler;
+                var time0 = scheduler.Clock.GetCurrentDateTimeUtc();
+                var callee = new Callee(scheduler);
+
+                await scheduler.RunAsync(async () =>
+                {
+                    var result = await connection.RunWithRetriableTransactionAsync(transaction => callee.DatabaseWorkAsync(connection, transaction));
+                    Assert.Equal(1, result);
+                });
+
+                callee.AssertCallTimesInrange(new Tuple<DateTime, DateTime>(time0, time0));
```

Comment:
If you want to use ValueTuple rather than Tuple, I'm fine to add a dependency. I'm also happy to commit as it is, then I'll have a look at simplifying in another PR :)